semanticdb/semanticdb/semanticdb.proto

```diff
@@ -320,5 +320,55 @@ message Diagnostic {
 
 message Synthetic {
   Range range = 1;
-  TextDocument text = 2;
+  Tree tree = 2;
+}
```

Comment:
I agree that the current API is suboptimal when it comes to reconstructing a fully desugared tree. Let's open a ticket for that.

Reply:
Not sure how to phrase the ticket

Reply:
This may be an indication that the problem is too abstract for now. In that case, the ticket would probably be not worth it.